enterprise/backend/src/metabase_enterprise/metabot_v3/client.clj

```diff
@@ -204,3 +210,53 @@
                               (:status response) (:reason-phrase response))
                       {:request (assoc options :body body)
                        :response response})))))
+
+(defn- build-multipart-request-options
+  "Build request options for multipart upload."
+  [image]
+  (let [base-headers (dissoc (request-headers) "Content-Type")]
+    {:headers          base-headers
+     :multipart        [{:name "image"
+                         :content (:tempfile image)
+                         :filename (:filename image)}]
+     :follow-redirects true
+     :throw-exceptions false}))
+
+(mu/defn analyze-chart
+  "Ask the AI service to analyze a chart image."
+  [chart-data :- [:map
+                 [:image_base64 :string]
+                 [:chart {:optional true} [:map
+                          [:name {:optional true} [:maybe :string]]
+                          [:description {:optional true} [:maybe :string]]]]
+                 [:timeline_events {:optional true} [:sequential [:map
+                                                                 [:name :string]
+                                                                 [:description {:optional true} [:maybe :string]]
+                                                                 [:timestamp :string]]]]]]
+  (let [url (analyze-chart-endpoint)
+        options (build-request-options chart-data)
+        response (post! url options)]
+    (if (= (:status response) 200)
+      (:body response)
+      (throw (ex-info (format "Error in request to AI service: unexpected status code: %d %s"
```

Comment:
It's not a new thing, but we might start including the type of the request in these error messages.